app/src/main/java/ch/admin/bag/dp3t/inform/ThankYouFragment.java

```diff
@@ -27,12 +34,32 @@ public ThankYouFragment() {
 		super(R.layout.fragment_thank_you);
 	}
 
+	@Override
+	public void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		secureStorage = SecureStorage.getInstance(getContext());
+	}
+
 	@Override
 	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
 		super.onViewCreated(view, savedInstanceState);
 
 		((InformActivity) requireActivity()).allowBackButton(false);
 
+		// Show the onset date in the thank you message
+		TextView thankYouTextView = view.findViewById(R.id.inform_thank_you_text);
+		long onsetDateInMillis = secureStorage.getPositiveReportOnsetDate();
+		if (onsetDateInMillis > 0L) {
+			String formattedDate = DateUtils.getFormattedDateWrittenMonth(onsetDateInMillis);
+			String thankYouText = getString(R.string.inform_send_thankyou_text_onsetdate).replace("{ONSET_DATE}", formattedDate);
+			int start = thankYouText.indexOf(formattedDate);
+			int end = thankYouText.indexOf("\n", start); // Include the " - today" part of the line
```

Comment:
are we sure that this will work for all translations? or can we make this somehow more robust?

Reply:
The best solution is to split the text into three sections and not use a spannable to make the onset date bold but instead use the textview style